### ISSUE TITLE
Add support for connecting to existing web terminal sessions

### ISSUE THREAD

**Issue by code-asher** [coder/coder#15435]
## Problem

Currently, we kill web terminal sessions after five minutes of being disconnected, which either kills any running processes or leaves "ghost" processes (like our develop script does) and loses all scrollback as well.

This causes frustration for users when the connection closes unexpectedly (for example because the screensaver causes the browser to idle while they are away for lunch, or there was a network blip while you were busy looking at some other window) so when they come back to the web terminal they have to start all over again.

You can work around this manually by, for example, running something like `tmux`, but it is 1) not obvious these sessions are ephemeral and 2) onerous, especially considering we can trivially make this a nice experience out of the box.

## Solution

We could simply remove the timeout, but then there is a risk that sessions could accumulate indefinitely without the user being aware.  To combat this we should:

1. List existing sessions with a new endpoint
2. When clicking "terminal", show a dropdown that is populated by these sessions
3. Add a "new" button to the dropdown which generates a new session (equivalent to the current behavior)
4. Remove the five minute timeout
5. Add delete buttons to each session in the dropdown

## Other ideas

1. It could be nice to have the ability to name sessions.  We could have an edit icon that lets you (re-)name the session, and when creating a new session we could ask for the (optional) name.  Without a name, we use the session ID.

## Notes

Terminal sessions can be backed by screen (when installed) or a naive buffer, but this has no bearing on the implementation.  The above will work for both types of sessions.

**Comment by MrPeacockNLB** [comment]
Cool if this will work out of the box! But it should not interferre with both of supported multiplexer **tmux** and **zellij**.

**Comment by code-asher** [comment]
Yup, this works independently of whatever you are running in the terminal, whether that is a multiplexer or anything else.
